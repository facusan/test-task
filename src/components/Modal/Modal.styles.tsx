import styled from "styled-components";

export const ModalWrapper = styled.div`
  display: flex;
  flex-direction: column;
  height: 100%;
  width: 100%;
`;
export const HeaderWrapper = styled.div`
  border-bottom-style: solid;
  border-bottom-width: 1px;
  border-bottom-color: lightgray;
  padding: 40px 25px;
  font-weight: bold;
  font-family: sans-serif;
  font-size: large;
  display: flex;
  align-items: center;
  justify-content: space-between;
`;
export const BodyWrapper = styled.div`
  padding: 40px 25px;
  height: 100%;
`;
export const FooterWrapper = styled.div`
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: lightgray;
  padding: 20px 25px;
  font-weight: bold;
  font-family: sans-serif;
  display: flex;
  justify-content: center;
  align-items: center;
`;
export const CloseIconWrapper = styled.div`
  &:hover {
    cursor: pointer;
  }
`;
