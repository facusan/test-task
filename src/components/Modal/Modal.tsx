import styled from "styled-components";
import { Button } from "../Button/Button";

const ModalWrapper = styled.div`
  display: flex;
  flex-direction: column;
`;
const HeaderWrapper = styled.div`
  border-bottom-style: solid;
  border-bottom-width: 1px;
  border-bottom-color: lightgray;
  padding: 40px 25px;
  font-weight: bold;
  font-family: sans-serif;
  font-size: large;
`;
const BodyWrapper = styled.div`
  padding: 40px 25px;
`;
const FooterWrapper = styled.div`
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: lightgray;
  padding: 20px 25px;
  font-weight: bold;
  font-family: sans-serif;
  display: flex;
  justify-content: center;
`;
interface ModalProps {
  next: Function;
  cancel: Function;
  children?: React.ReactNode;
}
export default function Modal({ next, cancel, children }: ModalProps) {
  return (
    <ModalWrapper>
      <HeaderWrapper>{"Send Ether"}</HeaderWrapper>
      <BodyWrapper>{children}</BodyWrapper>
      <FooterWrapper>
        <Button color={"secondary"} size={"small"}>
          Cancel
        </Button>
        <Button color={"primary"} size={"small"}>
          Next
        </Button>
      </FooterWrapper>
    </ModalWrapper>
  );
}
