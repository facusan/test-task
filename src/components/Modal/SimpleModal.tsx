import { Button } from "../Button/Button";
import { BodyWrapper, FooterWrapper, ModalWrapper } from "./Modal.styles";

interface SimpleModalProps {
  clickLabel: string;
  click: Function;
  children?: React.ReactNode;
}

export default function SimpleModal({
  click,
  clickLabel,
  children,
}: SimpleModalProps) {
  const onClick = () => {
    click();
  };
  return (
    <ModalWrapper>
      <BodyWrapper>{children}</BodyWrapper>
      <FooterWrapper>
        <Button color={"secondary"} onClick={onClick}>
          {clickLabel}
        </Button>
      </FooterWrapper>
    </ModalWrapper>
  );
}
