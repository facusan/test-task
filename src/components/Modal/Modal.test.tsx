import renderer from "react-test-renderer";
import Modal from "./Modal";
import SimpleModal from "./SimpleModal";

it("renders Modal correctly", () => {
  const component = renderer.create(
    <Modal cancel={() => {}} close={() => {}} next={() => {}}>
      <label>Modal body</label>
    </Modal>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders SimpleModal correctly", () => {
  const component = renderer.create(
    <SimpleModal click={() => {}} clickLabel="click">
      <label>Simple Modal body</label>
    </SimpleModal>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
