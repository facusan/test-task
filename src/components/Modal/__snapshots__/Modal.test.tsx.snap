// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Modal correctly 1`] = `
<div
  className="sc-gsnTZi IKqRv"
>
  <div
    className="sc-dkzDqf iaJygW"
  >
    <label>
      Send Ether
    </label>
    <div
      className="sc-jSMfEi hkeAGj"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="25px"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="25px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M7,7 L17,17 M7,17 L17,7"
          fill="none"
          stroke="#000"
          strokeWidth="2"
        />
      </svg>
    </div>
  </div>
  <div
    className="sc-hKMtZM bmNRYT"
  >
    <label>
      Modal body
    </label>
  </div>
  <div
    className="sc-eCYdqJ fZaxcr"
  >
    <button
      className="sc-bczRLJ lhoYHb"
      color="secondary"
      onClick={[Function]}
      size="small"
    >
      Cancel
    </button>
    <button
      className="sc-bczRLJ fvOGQT"
      color="primary"
      onClick={[Function]}
      size="small"
    >
      Next
    </button>
  </div>
</div>
`;

exports[`renders SimpleModal correctly 1`] = `
<div
  className="sc-gsnTZi IKqRv"
>
  <div
    className="sc-hKMtZM bmNRYT"
  >
    <label>
      Simple Modal body
    </label>
  </div>
  <div
    className="sc-eCYdqJ fZaxcr"
  >
    <button
      className="sc-bczRLJ jclefq"
      color="secondary"
      onClick={[Function]}
    >
      click
    </button>
  </div>
</div>
`;
